src/gsi/cplr_wrf_netcdf_interface.f90

```diff
@@ -1022,6 +1033,116 @@ subroutine convert_netcdf_mass_wrf(this)
+                filename_lakemask="lakemask.bin"; iunit_lakemask=115; l_iunit_opened=.true.
+                inquire(unit=iunit_lakemask, opened=l_iunit_opened)
+                if(l_iunit_opened) iunit_lakemask=iunit_lakemask+5000
+                if(print_verbose) write(6,*)' I/O unit for writing lakemask: ', iunit_lakemask
+                open(iunit_lakemask,file=trim(adjustl(filename_lakemask)),form='unformatted')
+                write(iunit_lakemask)field2   !LAKEMASK   (1=lake, 0=water)
+                close(iunit_lakemask)
+             else
+                write(6,'(1x,A)')' No LAKEMASK data in firstguess data. No lakemask is used for HOWV mask. reset i_howv_mask=1'
+                i_howv_mask=1                              ! i_howv_mask is changed on pe 0, so it must be broadcast to all PEs.
+             endif  ! if error in reading lakemask
+          endif   ! i_howv_mask = 2
+       endif  ! i_howv_3dda (reading 2D wave height from netcdf-format background) 
+
+!      Reading 10-m Wind Gust (GUST) in firstguess
+       if(i_gust_3dda >= 1) then                   ! if gust is found in anavinfo
+          rmse_var='GUST'
+          call ext_ncd_get_var_info (dh1,trim(rmse_var),ndim1,ordering,staggering, &
+               start_index,end_index, WrfType, ierr    )
+          if(print_verbose)then
+             write(6,*)' rmse_var=',trim(rmse_var)
+             write(6,*)' ordering=',ordering
+             write(6,*)' WrfType,ierr=',WrfType,ierr
+             write(6,*)' ndim1=',ndim1
+             write(6,*)' staggering=',staggering
+             write(6,*)' start_index=',start_index
+             write(6,*)' end_index=',end_index
+          end if
+          if(ierr == 0) then
+             call ext_ncd_read_field(dh1,DateStr1,TRIM(rmse_var),              &
+                  field2,WRF_REAL,0,0,0,ordering,           &
+                  staggering, dimnames ,               &
+                  start_index,end_index,               & !dom
+                  start_index,end_index,               & !mem
+                  start_index,end_index,               & !pat
+                  ierr                                 )
+             if(print_verbose)write(6,*)'convert_netcdf_mass_wrf:: max,min ',trim(adjustl(rmse_var)),'=',maxval(field2),minval(field2)
+             write(iunit)field2   !GUST  (2D Wind Gust)
+             i_gust_3dda = 2               ! gust was found both in anavinfo and firstguess
+          else
+             derr_msg='Warning: Error when get info in firstguess for var : '//trim(rmse_var)//    &
+                      ' ==> Stop GSI analysis ...  ierr='
+!            write(6,'(1x,A,A2,A,1x,I6)')trim(adjustl(myname_)),'::',trim(adjustl(derr_msg)),ierr
+!            call stop2(ierr)
+             call die(trim(adjustl(myname_)),trim(adjustl(derr_msg)),ierr)
+             i_gust_3dda = 0               ! skipping analysis of this variable
```

Comment:
Similar to comment above.

Reply:
line "i_gust_3dda = 0" is moved before call die.